standard/lexical-structure.md

```diff
@@ -324,64 +340,94 @@ Multiple translations are not performed. For instance, the string literal `"\u00
     ;
 
-Identifier_Part_Character
+fragment Identifier_Part_Character
     : Letter_Character
     | Decimal_Digit_Character
     | Connecting_Character
     | Combining_Character
     | Formatting_Character
     ;
 
-Letter_Character
-    : [\p{L}\p{Nl}]     // category Letter, all subcategories; category Number, subcategory letter
-    | Unicode_Escape_Sequence { IsLetterCharacter() }?
+fragment Letter_Character
+    : [\p{L}\p{Nl}]           // category Letter, all subcategories; category Number, subcategory letter
+    | Unicode_Escape_Sequence // only escapes for categories L & Nl allowed, see note below
+         { IsLetterCharacter() }?
     ;
 
-Combining_Character
-    : [\p{Mn}\p{Mc}]    // category Mark, subcategories non-spacing and spacing combining
-    | Unicode_Escape_Sequence { IsCombiningCharacter() }?
+fragment Combining_Character
+    : [\p{Mn}\p{Mc}]          // category Mark, subcategories non-spacing and spacing combining
+    | Unicode_Escape_Sequence // only escapes for categories Mn & Mc allowed, see note below
+         { IsCombiningCharacter() }?
     ;
 
-Decimal_Digit_Character
-    : [\p{Nd}]      // category Number, subcategory decimal digit
-    | Unicode_Escape_Sequence { IsDecimalDigitCharacter() }?
+fragment Decimal_Digit_Character
+    : [\p{Nd}]                // category Number, subcategory decimal digit
+    | Unicode_Escape_Sequence // only escapes for category Nd allowed, see note below
+         { IsDecimalDigitCharacter() }?
     ;
 
-Connecting_Character
-    : [\p{Pc}]      // category Punctuation, subcategory connector
-    | Unicode_Escape_Sequence { IsConnectingCharacter() }?
+fragment Connecting_Character
+    : [\p{Pc}]                // category Punctuation, subcategory connector
+    | Unicode_Escape_Sequence // only escapes for category Pc allowed, see note below
+         { IsConnectingCharacter() }?
     ;
 
-Formatting_Character
-    : [\p{Cf}]      // category Other, subcategory format
-    | Unicode_Escape_Sequence { IsFormattingCharacter() }?
+fragment Formatting_Character
+    : [\p{Cf}]                // category Other, subcategory format
+    | Unicode_Escape_Sequence // only escapes for category Cf allowed, see note below
+         { IsFormattingCharacter() }?
     ;
 ```
 
-> *Note*: For information on the Unicode character classes mentioned above, see *The Unicode Standard*. *end note*
+> *Note*:
+
+> 1. For information on the Unicode character classes mentioned above, see *The Unicode Standard*. *end note*
+
+> 2. The ANTLR semantic predicates above: `IsLetterCharacter`, `IsCombiningCharacter`, `IsDecimalDigitCharacter`, `IsConnectingCharacter` and `IsFormattingCharacter`; are informative *only*. How a compiler enforces the restriction on the allowable
+*Unicode_Escape_Sequence* values is an implementation issue. *end note*
+
+> 3. The explicit rules for identifiers depend on the implicit rules introduced by literal strings in the grammar:
```

Comment:
This attempts to clarify how the implicit and explicit rules interact to tokenise identifiers and [contextual] keywords, does it? Please review carefully.